Sources/SPMBuildCore/BinaryTarget+Extensions.swift

```diff
@@ -89,14 +89,29 @@ fileprivate extension Triple {
     }
 }
 
-fileprivate extension Triple.OS {
-    /// Returns a representation of the receiver that can be compared with platform strings declared in an XCFramework.
-    var asXCFrameworkPlatformString: String? {
-        switch self {
-        case .darwin, .linux, .wasi, .windows, .openbsd:
-            return nil // XCFrameworks do not support any of these platforms today.
-        case .macOS:
-            return "macos"
+fileprivate extension Triple {
+    var asXCFrameworkPlatformString: (String, String)? {
+        switch (os, abi) {
+        case (.macOS, .unknown):
+            return ("macos", "")
+        case (.ios, .other(name: "macabi")), (.ios, .other(name: "maccatalyst")):
+            return ("ios", "maccatalyst")
+        case (.ios, .unknown):
+            return ("ios", "")
+        case (.ios, .other(name: "simulator")):
+            return ("ios", "simulator")
+        case (.tvos, .unknown):
+            return ("tvos", "")
+        case (.tvos, .other(name: "simulator")):
+            return ("tvos", "simulator")
+        case (.watchos, .unknown):
+            return ("watchos", "")
+        case (.watchos, .other(name: "simulator")):
+            return ("watchos", "simulator")
+        case (.driverkit, .unknown):
+            return ("driverkit", "")
+        default:
```

Comment:
I think we should continue to be explicit here, so that we get compiler errors when new platforms are added.